examples/server/server.cpp

```diff
@@ -301,12 +301,43 @@ struct llama_client_slot
     }
 
     void print_timings() const {
-        LOG_TEE("\n");
-        LOG_TEE("%s: prompt eval time = %10.2f ms / %5d tokens (%8.2f ms per token, %8.2f tokens per second)\n",
-            __func__, t_prompt_processing, num_prompt_tokens_processed, t_prompt_processing / num_prompt_tokens_processed, 1e3 / t_prompt_processing * num_prompt_tokens_processed);
-        LOG_TEE("%s:        eval time = %10.2f ms / %5d runs   (%8.2f ms per token, %8.2f tokens per second)\n",
-            __func__, t_token_generation, n_decoded,t_token_generation / n_decoded, 1e3 / t_token_generation * n_decoded);
-        LOG_TEE("%s:       total time = %10.2f ms\n", __func__, t_prompt_processing + t_token_generation);
+        static char buffer[512];
+        double t_token = t_prompt_processing / num_prompt_tokens_processed;
+        double n_tokens_second = 1e3 / t_prompt_processing * num_prompt_tokens_processed;
+        sprintf(buffer, "prompt eval time     = %10.2f ms / %5d tokens (%8.2f ms per token, %8.2f tokens per second)",
+                t_prompt_processing, num_prompt_tokens_processed,
+                t_token, n_tokens_second);
+        LOG_INFO(buffer, {
+            {"slot_id",                     id},
+            {"task_id",                     task_id},
+            {"t_prompt_processing",         t_prompt_processing},
+            {"num_prompt_tokens_processed", num_prompt_tokens_processed},
+            {"t_token",                     t_token},
+            {"n_tokens_second",             n_tokens_second},
```

Comment:
I think we can already get rid of the `sprintf`, as it's more like to be logged to stdout.

Also I think maybe it's better to consider `LOG_*` functions as `LOG_*(tag, addition_info)` instead of `LOG_*(info, addition_info)`?

For example, we already have something like this in server.cpp:

```
    LOG_INFO("build info", {{"build", LLAMA_BUILD_NUMBER},
                            {"commit", LLAMA_COMMIT}});
```

So on "print_timings" we can do like this for example:

```
        LOG_INFO("prompt eval time", {
            {"slot_id",            id},
            {"task_id",            task_id},
            {"t_total_milisecs", t_total_milisecs}, // replaces t_prompt_processing
            {"n_token",          n_token}, // replaces n_decoded
            {"t_token_milisecs",            t_token_milisecs}, // replaces t_token
            {"n_tokens_per_second",    n_tokens_per_second}, // replaces n_tokens_second, most of the time we only care about this info
        });
```

Reply:
I agree with both of you: we need both. I suggest to have "message" for human reading the console and labels for log parsers.
@ngxson as we already have `function` and `line` in the log output, I think it's enough as default tags.
@ggerganov I will introduce an option to choose which logger formatter to use, it will be json as default.